File: app/src/main/java/hr/caellian/notestream/gui/dialog/DialogCancelOk.java
package hr.caellian.notestream.gui.dialog;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import hr.caellian.notestream.R;

/**
 * Created by tinsv on 16/07/2017.
 */

public class DialogCancelOk extends Dialog {
    String title;
    String description;

    View.OnClickListener cancelListener;
    View.OnClickListener okListener;

    public DialogCancelOk(@NonNull Context context, String title, String description, View.OnClickListener cancel, View.OnClickListener ok) {
        super(context);
        this.title = title;
        this.description = description;
        this.cancelListener = cancel;
        this.okListener = ok;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_cancel_ok);



        ((TextView) findViewById(R.id.labelTitle)).setText(title);
        TextView labelDescription = findViewById(R.id.labelDescription);
        labelDescription.setText(description);
//        labelDescription.setJustify(true); // TODO: Uncomment if it gets ported to support library.

        findViewById(R.id.buttonCancel).setOnClickListener(cancelListener);
        findViewById(R.id.buttonOk).setOnClickListener(okListener);
    }
}

File: app/src/main/java/hr/caellian/notestream/gui/dialog/DialogCancelOk.kt
/*
 * Copyright (C) 2018 Tin Svagelj
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
 * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
 * persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
 * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
 * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package hr.caellian.notestream.gui.dialog

import android.app.Dialog
import android.content.Context
import android.os.Build
import android.os.Bundle
import android.text.Layout
import android.view.View
import android.widget.Button
import android.widget.TextView

import hr.caellian.notestream.R

class DialogCancelOk(context: Context, internal var title: String, internal var description: String, internal var cancelListener: View.OnClickListener, internal var okListener: View.OnClickListener) : Dialog(context) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_cancel_ok)

        (findViewById<TextView>(R.id.labelTitle)).text = title
        val labelDescription = findViewById<TextView>(R.id.labelDescription)
        labelDescription.text = description
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            labelDescription.justificationMode = Layout.JUSTIFICATION_MODE_INTER_WORD
        }

        findViewById<Button>(R.id.buttonCancel).setOnClickListener(cancelListener)
        findViewById<Button>(R.id.buttonOk).setOnClickListener(okListener)
    }
}
